ts/src/hyperliquid.ts

```diff
@@ -1053,10 +1103,13 @@ export default class hyperliquid extends Exchange {
 
     priceToPrecision (symbol: string, price): string {
         const market = this.market (symbol);
-        // https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/tick-and-lot-size
-        const result = this.decimalToPrecision (price, ROUND, 5, SIGNIFICANT_DIGITS, this.paddingMode);
-        const decimalParsedResult = this.decimalToPrecision (result, ROUND, market['precision']['price'], this.precisionMode, this.paddingMode);
-        return decimalParsedResult;
+        const priceStr = this.numberToString (price) + '.';
+        const integerPart = priceStr.split ('.')[0];
+        const integerDigits = integerPart.length;
+        const significantDigits = Math.max (5, integerDigits);
```

Comment:
why do we need this logic here?

First we cap at 5 sig digits, then we apply the decimal places rounding, so here it seems we're trying to do everything at once ?

I think I would rather keep this logic unchanged

Reply:
It's because 123456 is now valid, since the network upgrade. Integer prices are not limited anymore by significantdigits.

Reply:
@gaardiolor 

1) - Got it

2), If you want a string length to work properly in PHP you should use it directly and not assign it in a statement, so when you need the str length just to `Math.max (5,integerPart.length)` 

3) Got it

Reply:
ok, php transpile issue fixed